qiskit_machine_learning/kernels/fidelity_quantum_kernel.py

```diff
@@ -0,0 +1,289 @@
+
+KernelIndices = List[Tuple[int, int]]
+
+
+class FidelityQuantumKernel(BaseKernel):
+    r"""
+    QuantumKernel
+
+    The general task of machine learning is to find and study patterns in data. For many
+    algorithms, the datapoints are better understood in a higher dimensional feature space,
+    through the use of a kernel function:
+
+    .. math::
+
+        K(x, y) = \langle f(x), f(y)\rangle.
+
+    Here K is the kernel function, x, y are n dimensional inputs. f is a map from n-dimension
+    to m-dimension space. :math:`\langle x, y \rangle` denotes the dot product.
+    Usually m is much larger than n.
+
+    The quantum kernel algorithm calculates a kernel matrix, given datapoints x and y and feature
+    map f, all of n dimension. This kernel matrix can then be used in classical machine learning
+    algorithms such as support vector classification, spectral clustering or ridge regression.
+
+    Here, the kernel function is defined as the overlap of two quantum states defined by a
+    parametrized quantum circuit (called feature map):
+
+    .. math::
+
+        K(x,y) = |\langle \phi(x) | \phi(y) \rangle|^2
+    """
+
+    def __init__(
+        self,
+        sampler: Sampler | None = None,
+        feature_map: QuantumCircuit | None = None,
+        fidelity: str | BaseStateFidelity = "zero_prob",
+        enforce_psd: bool = True,
+        evaluate_duplicates: str = "off_diagonal",
+    ) -> None:
+        """
+        Args:
+            sampler: An instance of the :class:`~qiskit.primitives.Sampler` primitive to be used to
+                evaluate fidelity. If ``None`` is given then a reference implementation will be
+                instantiated. If an instance of
+                :class:`~qiskit.algorithms.state_fidelities.BaseStateFidelity` is passed as
+                ``fidelity``, then the sampler value is not used.
+            feature_map: Parameterized circuit to be used as the feature map. If ``None`` is given,
+                :class:`~qiskit.circuit.library.ZZFeatureMap` is used with two qubits. If there's
+                a mismatch in the number of qubits of the feature map and the number of features
+                in the dataset, then the kernel will try to adjust the feature map to reflect the
+                number of features.
+            fidelity: An instance of the
+                :class:`~qiskit.algorithms.state_fidelities.BaseStateFidelity` primitive or a string
+                value defining a fidelity algorithm. Currently, the only supported string value
+                is ``zero_prob`` which corresponds to the
+                :class:`~qiskit.algorithms.state_fidelities.ComputeUncompute` fidelity.
+            enforce_psd: Project to closest positive semidefinite matrix if ``x = y``.
+                Default ``True``.
+            evaluate_duplicates: Defines a strategy how kernel matrix elements are evaluated if
+               duplicate samples are found. Possible values are:
```

Comment:
Would it make sense for evaluate_duplicates to be part of base class like enforce psd is? I see a comment about it being missed to start with from the trainable version.

Reply:
This would restrict the type of kernels to be implemented to those where  `k(x,x) = 1` is normalized, which I don't think always needs to be the case.

Reply:
I'd agree with @gentinettagian.